File: app/src/main/java/br/com/vicentec12/mygames/data/source/console/ConsoleDataSource.java
package br.com.vicentec12.mygames.data.source.console;

import java.util.List;

import br.com.vicentec12.mygames.data.model.Console;
import br.com.vicentec12.mygames.data.model.ConsoleWithGames;
import br.com.vicentec12.mygames.interfaces.Callbacks;

public interface ConsoleDataSource {

    interface OnConsolesWithGamesListedCallback {

        void onSucess(int message, List<ConsoleWithGames> consolesWithGames);

        void onErro(int message);

    }

    interface OnConsolesListedCallback {

        void onSucess(int message, List<Console> consoles);

        void onErro(int message);

    }

    void listConsoles(OnConsolesListedCallback callback);

    void listConsolesWithGames(OnConsolesWithGamesListedCallback callback);

    void insertConsoles(List<Console> consoles, Callbacks.OnLocalCallback callback);

    void deleteConsole(Console console, Callbacks.OnLocalCallback callback);

    void deleteAllConsoles(Callbacks.OnLocalCallback callback);

}

File: app/src/main/java/br/com/vicentec12/mygames/data/source/console/ConsoleDataSource.kt
package br.com.vicentec12.mygames.data.source.console

import br.com.vicentec12.mygames.data.model.Console
import br.com.vicentec12.mygames.data.model.ConsoleWithGames
import br.com.vicentec12.mygames.data.source.Result

interface ConsoleDataSource {

    suspend fun list(): Result<List<Console>>

    suspend fun listWithGames(): Result<List<ConsoleWithGames>>

    suspend fun insert(mConsoles: List<Console>): Result<Nothing>

    suspend fun delete(mConsole: Console): Result<Nothing>

    suspend fun deleteAll(): Result<Nothing>

}